src/derivative_operators/BC_operators.jl

```diff
@@ -1,11 +1,15 @@
-abstract type AbstractBC{T} <: AbstractDiffEqLinearOperator{T} end
+abstract type AbstractBC{T} end#<: AbstractDiffEqLinearOperator{T} end
 
+# Deepen type tree to support multi layered BCs in the future - a better version of PeriodicBC for example
+abstract type SingleLayerBC{T} <: AbstractBC{T} end
+abstract type MultiDimensionalBC{T, N} <: AbstractBC{T} end
+abstract type AbstractBoundaryPaddedArray{T, N} <: AbstractArray{T, N} end
 """
 Robin, General, and in general Neumann and Dirichlet BCs are all affine opeartors, meaning that they take the form Qx = Qax + Qb.
 """
-abstract type AffineBC{T,V} <: AbstractBC{T} end
+abstract type AffineBC{T,V} <: SingleLayerBC{T} end
 
-struct PeriodicBC{T} <: AbstractBC{T}
+struct PeriodicBC{T} <: SingleLayerBC{T}
```

Comment:
It can be `PeriodicBC{T,N}` for multi-dimensional. Then just work out how to build the padded arrays.

Reply:
That would work well as a special case

Reply:
added